lib/jobs/ipmi-job.js

```diff
@@ -127,34 +129,62 @@ function ipmiJobFactory(
-                data[cmd] = result;
-                if(data.password) {
-                    delete data.password;
-                }
-                return self._publishIpmiCommandResult(self.routingKey, cmd, data);
-            }).then(function() {
-                return waterline.workitems.findOne({ id: data.workItemId });
-            })
-            .then(function(workitem) {
-                    return waterline.workitems.setSucceeded(null, workitem);
-            })
-            .catch(function (err) {
-                if(data.password) {
-                    delete data.password;
-                }
-                logger.error("Failed to capture IPMI " + cmd +" status data.", {
-                    data: data,
-                    error: err
-                });
-                return waterline.workitems.findOne({ id: data.workItemId })
+                .then(function(result) {
+                    data[cmd] = result;
+                    if(data.password) {
+                        delete data.password;
+                    }
+                    return self._publishIpmiCommandResult(self.routingKey, cmd, data);
+                }).then(function() {
+                    return waterline.workitems.findOne({ id: data.workItemId });
+                })
                 .then(function(workitem) {
-                    return waterline.workitems.setFailed(null, workitem);
+                    var workitemOld = workitem;
+                    return Promise.resolve()
+                        .then(function(){
+                            if (workitemOld.failureCount > 0) {
+                                return protocolEvents.publishNodeAlert(workitemOld.node,
+                                    {nodeId: workitemOld.node,
+                                        obmType: "ipmi-obm-service",
+                                        state: "accessible",
+                                        command: cmd});
+                            }
+                        })
+                        .then(function(){
+                            return waterline.workitems.setSucceeded(null, workitemOld);
+                        });
+                })
+                .catch(function (err) {
+                    if(data.password) {
+                        delete data.password;
+                    }
+                    logger.error("Failed to capture IPMI " + cmd +" status data.", {
+                        data: data,
+                        error: err
+                    });
+                    return waterline.workitems.findOne({ id: data.workItemId })
+                        .then(function(workitem) {
+                            var workitemOld = workitem;
+                            return Promise.resolve()
+                                .then(function(){
+                                    //failure count ascends from 1
+                                    if (workitemOld.failureCount === 1) {
```

Comment:
so from succeed  to fail , orkitemOld.failureCount has been increased before this place?